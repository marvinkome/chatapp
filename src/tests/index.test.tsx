import React from 'react';
import { shallow } from 'enzyme';

import App from '../index';

describe('app unit tests', () => {
    it('renders correctly with defaults', () => {
        const app = shallow(<App />);
        expect(app).toMatchSnapshot();
    });
});
