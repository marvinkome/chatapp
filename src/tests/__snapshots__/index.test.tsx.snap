// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app tests renders correctly with defaults 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "height": 1334,
      },
      Object {
        "backgroundColor": "#fefefe",
      },
    ]
  }
>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
      ]
    }
  >
    <View
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Array [
          Object {
            "flex": 1,
            "flexDirection": "column-reverse",
          },
          Object {
            "backgroundColor": "#000",
          },
        ]
      }
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          pointerEvents="auto"
          style={
            Object {
              "backgroundColor": "#fafafa",
              "bottom": 0,
              "left": 0,
              "marginTop": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "shadowColor": "black",
              "shadowOffset": Object {
                "height": 0,
                "width": 0,
              },
              "shadowOpacity": 0.2,
              "shadowRadius": 5,
              "top": 0,
              "transform": Array [
                Object {
                  "translateX": 0,
                },
                Object {
                  "translateY": 0,
                },
              ],
            }
          }
        >
          <RCTScrollView
            automaticallyAdjustContentInsets={false}
            contentContainerStyle={
              Array [
                Object {
                  "padding": undefined,
                },
                undefined,
              ]
            }
            contentInset={
              Object {
                "bottom": 0,
              }
            }
            enableAutomaticScroll={true}
            enableResetScrollToCoords={true}
            extraHeight={75}
            extraScrollHeight={0}
            getScrollResponder={[Function]}
            handleOnScroll={[Function]}
            keyboardDismissMode="interactive"
            keyboardOpeningTime={250}
            keyboardShouldPersistTaps="handled"
            keyboardSpace={0}
            onScroll={[Function]}
            resetKeyboardSpace={[Function]}
            resetScrollToCoords={
              Object {
                "x": 0,
                "y": 0,
              }
            }
            scrollEventThrottle={1}
            scrollForExtraHeightOnAndroid={[Function]}
            scrollToEnd={[Function]}
            scrollToFocusedInput={[Function]}
            scrollToPosition={[Function]}
            showsVerticalScrollIndicator={true}
            style={
              Object {
                "backgroundColor": "transparent",
                "flex": 1,
              }
            }
            viewIsInsideTabBar={false}
          >
            <View>
              <RCTScrollView
                closeOnRowBeginSwipe={false}
                closeOnRowPress={true}
                closeOnScroll={true}
                dataArray={
                  Array [
                    <React.Fragment>
                      <Styled(Thumbnail)
                        size={80}
                        source={
                          Object {
                            "testUri": "../../../static/default-pic.png",
                          }
                        }
                        square={true}
                      />
                      <Styled(Body)>
                        <Styled(Text)
                          numberOfLines={1}
                        >
                          James Bond
                        </Styled(Text)>
                        <Styled(Text)
                          note={true}
                          numberOfLines={1}
                        >
                          Doing what you like..
                        </Styled(Text)>
                      </Styled(Body)>
                      <Styled(Text)
                        note={true}
                        style={
                          Object {
                            "fontSize": 13,
                          }
                        }
                      >
                        13:47pm
                      </Styled(Text)>
                    </React.Fragment>,
                    <React.Fragment>
                      <Styled(Thumbnail)
                        size={80}
                        source={
                          Object {
                            "testUri": "../../../static/default-pic.png",
                          }
                        }
                        square={true}
                      />
                      <Styled(Body)>
                        <Styled(Text)
                          numberOfLines={1}
                        >
                          James Cameroon
                        </Styled(Text)>
                        <Styled(Text)
                          note={true}
                          numberOfLines={1}
                        >
                          Doing what you like when you like
                        </Styled(Text)>
                      </Styled(Body)>
                      <Styled(Text)
                        note={true}
                        style={
                          Object {
                            "fontSize": 13,
                          }
                        }
                      >
                        17/06/2017
                      </Styled(Text)>
                    </React.Fragment>,
                  ]
                }
                dataSource={
                  ListViewDataSource {
                    "items": 2,
                  }
                }
                directionalDistanceChangeThreshold={2}
                disableLeftSwipe={false}
                disableRightSwipe={false}
                enableEmptySections={true}
                leftOpenValue={0}
                previewFirstRow={false}
                recalculateHiddenLayout={false}
                renderRow={[Function]}
                renderScrollComponent={[Function]}
                rightOpenValue={0}
                style={Object {}}
                swipeToOpenPercent={50}
              >
                <View>
                  <View
                    accessible={true}
                    isTVSelectable={true}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={null}
                  >
                    <View
                      onPress={[Function]}
                      style={
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "transparent",
                          "borderBottomWidth": 0.5,
                          "borderColor": "#c9c9c9",
                          "flexDirection": "row",
                          "marginLeft": 18,
                          "paddingBottom": 15,
                          "paddingRight": 18,
                          "paddingTop": 15,
                        }
                      }
                    >
                      <Image
                        size={80}
                        source={
                          Object {
                            "testUri": "../../../static/default-pic.png",
                          }
                        }
                        square={true}
                        style={
                          Object {
                            "borderRadius": 0,
                            "height": 56,
                            "width": 56,
                          }
                        }
                      />
                      <View
                        style={
                          Object {
                            "alignItems": null,
                            "alignSelf": null,
                            "flex": 1,
                          }
                        }
                      >
                        <Text
                          accessible={true}
                          allowFontScaling={true}
                          ellipsizeMode="tail"
                          numberOfLines={1}
                          style={
                            Object {
                              "color": "#000",
                              "fontFamily": "Roboto",
                              "fontSize": 16,
                              "marginLeft": 12,
                              "marginRight": 12,
                            }
                          }
                          uppercase={false}
                        >
                          James Bond
                        </Text>
                        <Text
                          accessible={true}
                          allowFontScaling={true}
                          ellipsizeMode="tail"
                          note={true}
                          numberOfLines={1}
                          style={
                            Object {
                              "color": "#808080",
                              "fontFamily": "Roboto",
                              "fontSize": 14,
                              "fontWeight": "200",
                              "marginLeft": 12,
                              "marginRight": 12,
                            }
                          }
                          uppercase={false}
                        >
                          Doing what you like..
                        </Text>
                      </View>
                      <Text
                        accessible={true}
                        allowFontScaling={true}
                        ellipsizeMode="tail"
                        note={true}
                        style={
                          Array [
                            Object {
                              "alignSelf": "center",
                              "color": "#808080",
                              "fontFamily": "Roboto",
                              "fontSize": 14,
                              "fontWeight": "200",
                            },
                            Object {
                              "fontSize": 13,
                            },
                          ]
                        }
                        uppercase={false}
                      >
                        13:47pm
                      </Text>
                    </View>
                  </View>
                  <View
                    accessible={true}
                    isTVSelectable={true}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={null}
                  >
                    <View
                      onPress={[Function]}
                      style={
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "transparent",
                          "borderBottomWidth": 0.5,
                          "borderColor": "#c9c9c9",
                          "flexDirection": "row",
                          "marginLeft": 18,
                          "paddingBottom": 15,
                          "paddingRight": 18,
                          "paddingTop": 15,
                        }
                      }
                    >
                      <Image
                        size={80}
                        source={
                          Object {
                            "testUri": "../../../static/default-pic.png",
                          }
                        }
                        square={true}
                        style={
                          Object {
                            "borderRadius": 0,
                            "height": 56,
                            "width": 56,
                          }
                        }
                      />
                      <View
                        style={
                          Object {
                            "alignItems": null,
                            "alignSelf": null,
                            "flex": 1,
                          }
                        }
                      >
                        <Text
                          accessible={true}
                          allowFontScaling={true}
                          ellipsizeMode="tail"
                          numberOfLines={1}
                          style={
                            Object {
                              "color": "#000",
                              "fontFamily": "Roboto",
                              "fontSize": 16,
                              "marginLeft": 12,
                              "marginRight": 12,
                            }
                          }
                          uppercase={false}
                        >
                          James Cameroon
                        </Text>
                        <Text
                          accessible={true}
                          allowFontScaling={true}
                          ellipsizeMode="tail"
                          note={true}
                          numberOfLines={1}
                          style={
                            Object {
                              "color": "#808080",
                              "fontFamily": "Roboto",
                              "fontSize": 14,
                              "fontWeight": "200",
                              "marginLeft": 12,
                              "marginRight": 12,
                            }
                          }
                          uppercase={false}
                        >
                          Doing what you like when you like
                        </Text>
                      </View>
                      <Text
                        accessible={true}
                        allowFontScaling={true}
                        ellipsizeMode="tail"
                        note={true}
                        style={
                          Array [
                            Object {
                              "alignSelf": "center",
                              "color": "#808080",
                              "fontFamily": "Roboto",
                              "fontSize": 14,
                              "fontWeight": "200",
                            },
                            Object {
                              "fontSize": 13,
                            },
                          ]
                        }
                        uppercase={false}
                      >
                        17/06/2017
                      </Text>
                    </View>
                  </View>
                </View>
              </RCTScrollView>
            </View>
          </RCTScrollView>
        </View>
      </View>
      <View
        onLayout={[Function]}
        pointerEvents="box-none"
      >
        <View
          onLayout={[Function]}
        >
          <View
            style={
              Object {
                "backgroundColor": "#3F51B5",
                "borderBottomColor": "#3F51B5",
                "borderBottomWidth": 0.5,
                "elevation": 3,
                "flexDirection": "row",
                "height": NaN,
                "justifyContent": "center",
                "left": 0,
                "paddingLeft": 10,
                "paddingRight": 10,
                "paddingTop": 18,
                "right": 0,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.2,
                "shadowRadius": 1.2,
                "top": 0,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "flex-start",
                  "alignSelf": "center",
                  "flex": 1,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                numberOfLines={1}
                style={
                  Object {
                    "color": "#FFF",
                    "fontFamily": "Roboto",
                    "fontSize": 19,
                    "fontWeight": "700",
                    "marginLeft": undefined,
                    "paddingLeft": 4,
                    "paddingTop": 1,
                    "textAlign": "center",
                  }
                }
              >
                Home
              </Text>
            </View>
            <View
              style={
                Object {
                  "alignItems": "flex-end",
                  "alignSelf": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "flex-end",
                }
              }
            />
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
